### ISSUE TITLE
export ArraySize from hyperformula

### ISSUE THREAD

**Issue by MartinDawson** [handsontable/hyperformula#843]
ArraySize is needed in customFunctions _I think_.

Here's my custom funciton below. You can see it accepts either 1, 2, 3 or 4 arguments.

If the first parameter is `financialStatements` then I need to tell hyperformula to return an ArraySize so that it can overflow into subsequent cells.

```js
  // Pre-fixed with TT due to there already being a FinancialPlugin
  // in hyperformula
  class TTFinancialPlugin extends FunctionPlugin {
    financial({ args }) {
      if (!args.length) {
        return new InvalidArgumentsError(1);
      }
      if (!hasFinancialsLoaded) {
        return "Loading...";
      }
      const attribute = args[0].value;
      // TODO: Add proper error checking here later
      if (args.length === 1) {
        if (attribute === "currencyCode") {
          const currencyCode = ttmData[attribute];
          return convertSubCurrencyToCurrency(currencyCode);
        }
        if (attribute === "financialStatements") {
          return SimpleRangeValue.onlyValues(statements);
        }
        return ttmData[attribute] ?? "";
      }
      const startDate = args[1].value;
      const statementType = getTypeOfStatementToUse(attribute);
      if (args.length === 2) {
        if (attribute === "description") {
          return ttmData[attribute];
        }
        return (
          historicalDataArrays[statementType].yearly[startDate][attribute] ?? ""
        );
      }
      const endDate = args[2].value;
      if (args.length === 3) {
        return SimpleRangeValue.onlyValues([
          getYearlyValues(attribute, statementType, startDate, endDate),
        ]);
      }
    }
    financialSize({ args }) {
      if (!hasFinancialsLoaded) {
        return ArraySize.scalar();
      }

      const attribute = args[0].value;
      const statementType = getTypeOfStatementToUse(attribute);
      const startDate = args[1] ? args[1].value : null;
      const endDate = args[2] ? args[2].value : null;

      if (attribute === "financialStatements") {
        return ArraySize.fromArray(statements);
      }

      if (args.length === 3) {
        const yearlyValues = getYearlyValues(
          attribute,
          statementType,
          startDate,
          endDate,
        );

        return ArraySize.fromArray([yearlyValues]);
      }

      return ArraySize.scalar();
    }
  }

  TTFinancialPlugin.implementedFunctions = {
    FINANCIAL: {
      method: "financial",
      arraySizeMethod: "financialSize",
    },
  };

  TTFinancialPlugin.aliases = {
    FIN: "FINANCIAL",
  };

```

The above works well but I believe `ArraySize` needs to be exported from hyperformula because currently I have to import it like so:

```
import { ArraySize } from "hyperformula/es/ArraySize";
```

**Comment by mlconnor** [comment]
Could anyone suggest a workaround for the browser to export ArraySize until this is included properly?

**Comment by mlconnor** [comment]
Here is a sandbox. Take a peek at hyperformulaConfig.js file. It's broken because the ArraySize class isn't exported. This class is required to implement custom array functions because the sizing method requires the method to return an ArraySize object.

https://codesandbox.io/s/hyperformula-custom-fn-arraysize-issue-ywryj1

**Comment by sequba** [comment]
@mlconnor Thanks for your feedback. This bug is a known issue and will be fixed in one of the upcoming releases. I'm glad that you could find a workaround.

**Comment by sequba** [comment]
Will be done together with #779

**Comment by sequba** [comment]
Hi @MartinDawson, @mlconnor!

I'm more than happy to announce that we just released [HyperFormula 2.3.0](https://github.com/handsontable/hyperformula/releases/tag/2.3.0) where this issue is fixed. We are closing this issue as solved. If there is anything that won't work for you after updating please leave a comment.
